QuickBytes/src/main/java/com/quickbytes/controller/ItemController.java

```diff
@@ -0,0 +1,75 @@
+package com.quickbytes.controller;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.quickbytes.model.Item;
+import com.quickbytes.model.Vendor;
+import com.quickbytes.repository.ItemRepository;
+import com.quickbytes.repository.VendorRepository;
+
+@RestController
+public class ItemController {
+	@Autowired
+	ItemRepository itemRepository;
+	
+	@Autowired
+	VendorRepository vendorRepository;
+	
+	@PostMapping("/item/{vid}")
+	public void postItem(@RequestBody Item i,@PathVariable("vid") Long vid) {
+		//use JpaRepository Interface
+		//has lot of precreated methods for db interaction
+		Optional<Vendor> optionalVendor = vendorRepository.findById(vid);
+		if(!optionalVendor.isPresent()) {
+			throw new RuntimeException("invalid vendor id");
+		}
+		i.setVendor(optionalVendor.get());
+		itemRepository.save(i);
+	}
+	
+	@GetMapping("/items")
+	public List<Item> getAllItems() {
```

Comment:
Going back to post work done, it seems that the general format followed was the following:
public List<Item> getAllItems(@RequestBody Item item)

Verify whether or not this matters. If it tests and works then disregard.

Reply:
@requestbody is used for post and put, because it takes JSON input, it is not needed for get and delete because it does not need any input

Reply:
I see, just saw this. Thank you for teaching me that.